Survey/SurveyUnitTests/Sources/Specs/Login/LoginRepositorySpec.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  LoginRepositorySpec.swift
+//  SurveyUnitTests
+//
+//  Created by Taher on 14/11/22.
+//
+//  swiftlint:disable closure_body_length
+
+import Combine
+import Nimble
+import Quick
+
+@testable import Survey
+
+final class LoginRepositorySpec: QuickSpec {
+
+    override func spec() {
+
+        var loginRepository: LoginRepository!
+        var loginNetwork: MockLoginNetworkAPI!
+        var isResponseReceived = false
+        var cancellables = CancelBag()
+
+        describe("a LoginRepository") {
+
+            beforeEach {
+                loginNetwork = MockLoginNetworkAPI()
+                loginRepository = LoginRepository(network: loginNetwork)
+                isResponseReceived = false
+                executeRequest()
+            }
+
+            context("a call to getLoginToken") {
+
+                it("should receive  response") {
+                    expect(isResponseReceived)
+                        .toEventually(
+                            equal(true),
+                            timeout: .seconds(2)
+                        )
+                }
+            }
```

Comment:
This only tests that when `LoginRepository` calls its `getLoginToken()`, it receives the event from the dependency `loginNetwork`, but it doesn't test the result. The tests should be same like you do in `LoginNetworkAPISpec`, meaning they atleast contains two cases:
- When mocked network API returns failure -> the repository returns a corresponding failure.
- When mocked network API returns success -> the repository returns a corresponding success.